Update package manager instructions with community-hosted repos (#42022)
kubernetes/website: content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md

---
title: Changing The Kubernetes Package Repository
content_type: task
weight: 120
---

<!-- overview -->

This page explains how to switch from one Kubernetes package repository to another
when upgrading Kubernetes minor releases. Unlike deprecated Google-hosted
repositories, the Kubernetes package repositories are structured in a way that
there's a dedicated package repository for each Kubernetes minor version.

For more information about the Kubernetes community-owned package repositories,

> see ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/2023/08/15/pkgs-k8s-io-introduction/).

Reviewer: We'd better wait for confirmation of that publication date. Also, this is the wrong URL even if that date is right.

Author: Blog post link changed to `/blog/2023/08/15/pkgs-k8s-io-introduction/`

## {{% heading "prerequisites" %}}

This document assumes that you're already using the Kubernetes community-owned
package repositories. If that's not the case, it's strongly recommend to migrate
to the Kubernetes package repositories as described in

> ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/2023/08/15/pkgs-k8s-io-introduction/).

Reviewer: We'd better wait for confirmation of that publication date. Also, this is the wrong URL even if that date is right.

Author: Blog post link changed to `/blog/2023/08/15/pkgs-k8s-io-introduction/`

> ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/15/pkgs-k8s-io-introduction/).

Reviewer 2: ```suggestion
["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/22/pkgs-k8s-io-introduction/).
```

### Verifying if the Kubernetes package repositories are used

If you're unsure if you're using the Kubernetes package repositories or the
Google-hosted repository, take the following steps to verify:

{{< tabs name="k8s_install_versions" >}}
{{% tab name="Ubuntu, Debian or HypriotOS" %}}

Print the contents of the file that defines the Kubernetes `apt` repository:

```shell
# On your system, this configuration file could have a different name
pager /etc/apt/sources.list.d/kubernetes.list
```

If you see a line similar to:

```
deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v{{< skew currentVersionAddMinor -1 "." >}}/deb/ /
```

**You're using the Kubernetes package repositories and this guide applies to you.**
Otherwise, it's strongly recommend to migrate to the Kubernetes package repositories

> as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/2023/08/15/pkgs-k8s-io-introduction/).

Reviewer: We'd better wait for confirmation of that publication date. Also, this is the wrong URL even if that date is right.

Author: Blog post link changed to `/blog/2023/08/15/pkgs-k8s-io-introduction/`

> as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/15/pkgs-k8s-io-introduction/).

Reviewer 2: ```suggestion
as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/22/pkgs-k8s-io-introduction/).
```

{{% /tab %}}
{{% tab name="CentOS, RHEL or Fedora" %}}

Print the contents of the file that defines the Kubernetes `yum` repository:

```shell
# On your system, this configuration file could have a different name
cat /etc/yum.repos.d/kubernetes.repo
```

If you see `baseurl` similar to the `baseurl` in the output below:

```
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/v{{< skew currentVersionAddMinor -1 "." >}}/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/v{{< skew currentVersionAddMinor -1 "." >}}/rpm/repodata/repomd.xml.key
exclude=kubelet kubeadm kubectl
```

**You're using the Kubernetes package repositories and this guide applies to you.**
Otherwise, it's strongly recommend to migrate to the Kubernetes package repositories

> as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/2023/08/15/pkgs-k8s-io-introduction/).

Reviewer: We'd better wait for confirmation of that publication date. Also, this is the wrong URL even if that date is right.

Author: Blog post link changed to `/blog/2023/08/15/pkgs-k8s-io-introduction/`

> as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/15/pkgs-k8s-io-introduction/).

Reviewer 2: ```suggestion
as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/22/pkgs-k8s-io-introduction/).
```

---
title: Changing the Kubernetes package repository

> min-kubernetes-server-version: 1.24

Reviewer: Why this line?

content_type: task
weight: 120
---

<!-- overview -->

> This page explains how to switch from one to another Kubernetes package repository
> upon upgrading from one to another Kubernetes minor release. Unlike Google-hosted

Reviewer 3: ```suggestion
This page explains how to switch from one package repository to another
when upgrading Kubernetes minor releases. Unlike Google-hosted
```

repositories, the Kubernetes community-owned package repositories are structured
in a way that there's a dedicated package repository for each Kubernetes minor version.

For more information about the Kubernetes community-owned package repositories,
see the ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](TBD)
blog post.

> ## {{% heading "prerequisites" %}}

Reviewer 3: I think you can simplify the text by merging the _prerequisites_ and the first step.

Think about it this way: if the first step is to check if the repos are in use, then it is not really a prerequisite to start the procedure.

Reviewer: We should keep this heading. Maybe the first step becomes a subheading?

Author: > Maybe the first step becomes a subheading?

I like this idea!

This document assumes that you're already using the Kubernetes community-owned
package repositories. If that's not the case, it's strongly recommend to migrate
from the Google-hosted repository to the Kubernetes package repositories

> as described in [the blog post](TBD).

Reviewer: FIXME!

If you're unsure if you're using the Kubernetes package repositories or the
Google-hosted repository, the first step of this document explains how to check
what repository is in use.

<!-- steps -->

## Verifying if the Kubernetes package repositories are used

The purpose of this step is to verify if you're using the Kubernetes package repositories.

{{< tabs name="k8s_install_versions" >}}
{{% tab name="Ubuntu, Debian or HypriotOS" %}}

> Print contents of the file that defines the Kubernetes `apt` repository:

Reviewer 3: ```suggestion
Print the contents of the file that defines the Kubernetes `apt` repository:
```

```shell

> cat /etc/apt/sources.list.d/kubernetes.list

Reviewer: (nit / preference)
```suggestion
pager /etc/apt/sources.list.d/kubernetes.list
```

Debian provides `pager` as an abstraction over the user's preferred pager.

```

> If you see the URL similar to:

Reviewer: ```suggestion
If you see a line similar to:
```

```
deb https://pkgs.k8s.io/core:/stable:/v{{< skew currentVersionAddMinor -1 "." >}}/deb/ /
```

**You're using the Kubernetes package repositories and this guide applies to you.**
Otherwise, it's strongly recommend to migrate to the Kubernetes package repositories

> as described in [the blog post](TBD).

Reviewer: FIXME

Author: Well aware of it, I still didn't receive date for blog post which is part of the URL, but as soon as I do, I'll update this. :)

{{% /tab %}}
{{% tab name="CentOS, RHEL or Fedora" %}}

> Print contents of the file that defines the Kubernetes `yum` repository:

Reviewer 3: ```suggestion
Print the contents of the file that defines the Kubernetes `yum` repository:
```

```shell

> cat /etc/yum.repos.d/kubernetes.repo

Reviewer: ```suggestion
# On your system, this configuration file could have a different name
cat /etc/yum.repos.d/kubernetes.repo
````

```

If you see `baseurl` similar to the `baseurl` in the output below:

```
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/v{{< skew currentVersionAddMinor -1 "." >}}/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/v{{< skew currentVersionAddMinor -1 "." >}}/rpm/repodata/repomd.xml.key
exclude=kubelet kubeadm kubectl
```

**You're using the Kubernetes package repositories and this guide applies to you.**
Otherwise, it's strongly recommend to migrate to the Kubernetes package repositories
as described in [the blog post](TBD).

{{% /tab %}}
{{< /tabs >}}

{{< note >}}
The URL used for the Kubernetes package repositories is not limited to `pkgs.k8s.io`,
it can also be one of:

- `pkgs.k8s.io`
- `pkgs.kubernetes.io`
- `packages.kubernetes.io`
- `packages.kubernetes.io`
{{</ note >}}

## Switching to another Kubernetes package repository

This step should be done upon upgrading from one to another Kubernetes minor

> release in order to get access to packages for the desired Kubernetes minor

Reviewer 3: ```suggestion
release in order to get access to the packages of the desired Kubernetes minor
```

```

3. Save the file and exit your text editor. Proceed with following the relevant upgrade guidelines.

{{% /tab %}}
{{% tab name="CentOS, RHEL or Fedora" %}}

1. Open the file that defines the Kubernetes `yum` repository using a text editor of your choice:

```shell
nano /etc/yum.repos.d/kubernetes.repo
```

You should see a file with two URLs that contain your current Kubernetes
minor version. For example, if you're using v{{< skew currentVersionAddMinor -1 "." >}},
you should see this:

```
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/v{{< skew currentVersionAddMinor -1 "." >}}/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/v{{< skew currentVersionAddMinor -1 "." >}}/rpm/repodata/repomd.xml.key
exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni
```

2. Change the version in these URLs to **the next available minor release**, for example:

```
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/v{{< param "version" >}}/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/v{{< param "version" >}}/rpm/repodata/repomd.xml.key
exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni
```

> 3. Save the file and exit your text editor. Proceed with following the relevant upgrade guidelines.

Reviewer 3: ```suggestion
3. Save the file and exit your text editor. Continue following the relevant upgrade guidelines.
```
